src/utils/helpers.js

```diff
@@ -217,10 +243,9 @@ class Helpers {
  */
   static checkToken(req) {
     const { authorization } = req.headers;
-    if (!authorization) {
-      throw new ApiError(401, 'Access denied, Token required');
-    }
-    const token = req.headers.authorization.split(' ')[1] || req.headers.authorization || req.headers['x-access-token'] || req.headers.token || req.body.token;
+    const posibleOptions = (req.headers['access-token'] || req.headers.token || req.params.token || req.cookies.token || null);
+    const token = authorization ? authorization.split(' ')[1] : posibleOptions;
```

Comment:
Great job on your implementations so far @King-Success. I taught to point your attention to the fact that authorization may be an array or a string. The current implementation `authorization.split(' ')[1]` returns a value if an array of with a length of 2 exists and undefined if authorization is a string.

Reply:
Thanks for pointing that out, however, in my experience authorization headers are often set like so `Bearer user-token` or `Basic user-token` resulting in a string with exactly two parts separated by a single space. With that, the split method would return an array of length 2 containing both parts of the authorization header. The last part `[1]` would ensure that `token` variable receives the latter part of the authorization header which in this case is the actual user token.
[See this MDN doc](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization)

Reply:
I agree with you @King-Success. I meant to actually say a string of two parts separated by a space. As they are really no hard or fast rule to setting tokens, it may not be a bad idea to design a check for a wide range of possibilities as this enables interoperability. I have seen in my experience where authorization is set as a single string just for convenience and why not, there are no hard rules mandating otherwise. The `Bearer user-token` was actually introduced by W3C and it has been reused ever since by many websites but it isn't a rule and other implementations would still work. In our case, since we may not be interacting with external APIs and we have set up our web server to look in the cookies for the token, we may as well have ignored other possibilities and we would still have a working system but I am guessing for scalability the initial implementation was designed to look in as many places as possible.

Reply:
I am satisfied with your response though. Maybe the team can weigh in on this at a later time.